src/blueprint.zig

```diff
@@ -336,26 +339,28 @@ pub const Pattern = struct {
 	};
 
 	pub fn initFromString(allocator: NeverFailingAllocator, source: []const u8) !@This() {
-		var specifiers = std.mem.splitScalar(u8, source, ',');
+		var specifiers = std.mem.splitScalar(u8, source, expressionSeparator);
 		var totalWeight: f32 = 0;
 
 		var weightedEntries: ListUnmanaged(struct {block: Block, weight: f32}) = .{};
 		defer weightedEntries.deinit(main.stackAllocator);
 
 		while(specifiers.next()) |specifier| {
-			var iterator = std.mem.splitScalar(u8, specifier, '%');
+			var blockId = specifier;
+			var weight: f32 = 1.0;
+
+			if(std.mem.containsAtLeastScalar(u8, specifier, 1, weightSeparator)) {
+				var iterator = std.mem.splitScalar(u8, specifier, weightSeparator);
+				const weightString = iterator.first();
+				blockId = iterator.rest();
 
-			var weight: f32 = undefined;
-			var block = main.blocks.parseBlock(iterator.rest());
+				weight = std.fmt.parseFloat(f32, weightString) catch return error.@"Weight not a valid number";
+				if(weight <= 0) return error.@"Weight must be >= 0";
```

Comment:
Should be >, also is not intended for programmers, so better use english words instead
```suggestion
				if(weight <= 0) return error.@"Weight must be greater than 0";
```